api.go

```diff
@@ -45,7 +45,7 @@ func (engine *Engine) FetchNicAndVm() {
 
 func (engine *Engine) ObtainNic(nic_id string) error {
 
-	_, err := engine.Exo.AddIpToNic(nic_id, engine.ExoIP.String())
+	_, err := engine.Exo.AddIpToNic(nic_id, engine.ExoIP.String(), engine.AsyncInfo)
 	if err != nil {
```

Comment:
here `err` was representing the job creation, and not the addition of the IP to the NIC